sdk/healthinsights/azure-health-insights-radiologyinsights/src/samples/java/com/azure/health/insights/radiologyinsights/SampleCriticalResultInferenceAsync.java

```diff
@@ -0,0 +1,287 @@
+            + "\r\n20-year-old female presenting with abdominal pain. Surgical history significant for appendectomy."
+            + "\r\n "
+            + "\r\nCOMPARISON:   "
+            + "\r\nRight upper quadrant sonographic performed 1 day prior."
+            + "\r\n "
+            + "\r\nTECHNIQUE:   "
+            + "\r\nTransabdominal grayscale pelvic sonography with duplex color Doppler "
+            + "\r\nand spectral waveform analysis of the ovaries."
+            + "\r\n "
+            + "\r\nFINDINGS:   "
+            + "\r\nThe uterus is unremarkable given the transabdominal technique with "
+            + "\r\nendometrial echo complex within physiologic normal limits. The "
+            + "\r\novaries are symmetric in size, measuring 2.5 x 1.2 x 3.0 cm and the "
+            + "\r\nleft measuring 2.8 x 1.5 x 1.9 cm.\n \r\nOn duplex imaging, Doppler signal is symmetric."
+            + "\r\n "
+            + "\r\nIMPRESSION:   "
+            + "\r\n1. Normal pelvic sonography. Findings of testicular torsion."
+            + "\r\n\nA new US pelvis within the next 6 months is recommended."
+            + "\n\nThese results have been discussed with Dr. Jones at 3 PM on November 5 2020.\n "
+            + "\r\n";
+
+    /**
+     * The main method is the entry point for the application. It initializes and uses
+     * the RadiologyInsightsAsyncClient to perform Radiology Insights operations.
+     *
+     * @param args The command-line arguments passed to the program.
+     */
+    public static void main(final String[] args) throws InterruptedException {
+        String endpoint = Configuration.getGlobalConfiguration().get("AZURE_HEALTH_INSIGHTS_ENDPOINT");
+        String apiKey = Configuration.getGlobalConfiguration().get("AZURE_HEALTH_INSIGHTS_API_KEY");
+        
+        RadiologyInsightsAsyncClient radiologyInsightsAsyncClient = new RadiologyInsightsClientBuilder()
+                .endpoint(endpoint).serviceVersion(RadiologyInsightsServiceVersion.getLatest())
+                .credential(new AzureKeyCredential(apiKey)).buildAsyncClient();
+
+        PollerFlux<PollOperationDetails, RadiologyInsightsInferenceResult> asyncPoller = radiologyInsightsAsyncClient
+                .beginInferRadiologyInsights(createRadiologyInsightsRequest());
+        
+        asyncPoller.takeUntil(isComplete).filter(isComplete).subscribe(completedResult -> {
+            System.out.println("Completed poll response, status: " + completedResult.getStatus());
+            if (completedResult.getStatus() == LongRunningOperationStatus.SUCCESSFULLY_COMPLETED) {
+                mono = completedResult.getFinalResult();
+            } else {
+                System.out.println(
+                        "Poll Request Failed with message: " + completedResult.getValue().getError().getMessage());
+            }
+            latch.countDown();
+        });
+
+        latch.await();
+        displayCriticalResults(mono.block());
```

Comment:
The reactor code may not be best practise.

candidate
```java
        Mono<RadiologyInsightsInferenceResult> finalResult = asyncPoller.last().flatMap(pollResult -> {
            if (pollResult.getStatus() == LongRunningOperationStatus.SUCCESSFULLY_COMPLETED) {
                return pollResult.getFinalResult();
            } else {
                System.out.println(
                        "Poll Request Failed with message: " + pollResult.getValue().getError().getMessage());
                return Mono.empty();    // Mono.error ?
            }
        });

        finalResult.doFinally(() -> {
            latch.countDown();
        }).subscribe(result -> {
            displayCriticalResults(result);
        });

        latch.countDown();
```

Reply:
refactored to reflect suggested code.